sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2498,6 +2503,10 @@ class Dataset[T] private[sql](
    *
    * Use [[summary]] for expanded statistics and control over which statistics to compute.
    *
+   * @note String columns will be implicitly cast to double when performing operations like mean
```

Comment:
I think this documentation should rather go to `functions.scala` or the SQL documentation under `ExpressionDescription`. This seems like not only specific to this API alone. You can remove this here and do it separately in a separate PR.

Reply:
This method accept numeric and string, I guess this note help us understand the string stat output. For sepecific functions, the implicit casting happens not only for string only.

Reply:
But this applies to all other stddev and mean functions too:

```scala
scala> sql("select mean('1.0')").show()
+------------------------+
|avg(CAST(1.0 AS DOUBLE))|
+------------------------+
|                     1.0|
+------------------------+
```

I would locate the note in a better place to deduplicate the documentation.

Reply:
all right